src/js/graph/graph.js

```diff
@@ -95,93 +126,70 @@ Graph.prototype.removeNode = function(node) {
     node.removeGraph();
     this._graphImpl.removeNode(node.getId());
 };
-function logEdgeError(graph, sourceId, targetId, edgeType) {
-    var message;
-
-    if (!graph.getNodeById(sourceId) && !graph.getNodeById(targetId)) {
-        message = [
-            'Cannot create `',
-            edgeType,
-            '` edge between non-existent node ids `',
-            sourceId,
-            '` and `',
-            targetId,
-            '`.'
-        ].join('');
-    } else if (!graph.getNodeById(sourceId)) {
-        message = [
-            'Cannot create `',
-            edgeType,
-            '` edge from non-existent node id `',
-            sourceId,
-            '` to node id `',
-            targetId,
-            '`.'
-        ].join('');
-    } else if (!graph.getNodeById(targetId)) {
-        message = [
-            'Cannot create `',
-            edgeType,
-            '` edge from node id `',
-            sourceId,
-            '` to non-existent node id `',
-            targetId,
-            '`.'
-        ].join('');
+// Edge manipulation functions
+Graph.prototype.edgeExists = function(sourceId, targetId, edgeType) {
+    return this._graphImpl.hasEdge(sourceId, targetId, edgeType);
+};
+Graph.prototype.addEdge = function(sourceId, targetId, edgeType, toType) {
```

Comment:
I've moved addEdge & addEdges, and removeEdge to the _ namespace, but I think edgeExists ought to stay in the public namespace. Since it's non-destructive, checking the graph for an edge explicitly vs having to look up a node, then look up that node's edges seems like a reasonable convenience function to leave public on the graph.